hazelcast/src/test/java/com/hazelcast/jet/core/ExecutionLifecycleTest.java

```diff
@@ -184,6 +206,91 @@ public void when_pmsInitThrows_then_jobFails() {
         assertJobFailed(job, MOCK_ERROR);
     }
 
+    @Test
+    public void when_pmsInitThrowsNonSerializable_then_jobFails() {
+        // Given
+        SupplierEx<ProcessorSupplier> supplier = () ->  new MockPS(MockP::new, MEMBER_COUNT);
+        DAG dag = new DAG().vertex(new Vertex("test",
+                new MockPMS(supplier)
+                        .initThrowsNonSerializable()));
+
+        // When
+        Job job = runJobExpectFailure(dag, false);
+
+        // Then
+        assertPmsClosedWithError();
+        assertJobFailed(job, NON_SERIALIZABLE_EXCEPTION);
+    }
+
+    @Test
+    public void when_pmsNonCooperativeInitThrowsNonSerializable_then_jobFails()
+            throws InterruptedException, ExecutionException {
+        // Given
+        SupplierEx<ProcessorSupplier> supplier = () ->  new MockPS(MockP::new, MEMBER_COUNT);
+        DAG dag = new DAG().vertex(new Vertex("test",
+                new MockPMS(supplier)
+                        .initBlocks()
+                        .initThrowsNonSerializable()));
+
+        Future<Job> jobFuture = spawn(() -> newJob(dag));
+
+        MockPMS.waitBlockingSemaphore();
+
+        for (int i = 0; i < MEMBER_COUNT; i++) {
+            MockPMS.unblock();
+        }
+
+        //Wait for job to finish
+        assertThrows(CompletionException.class, () -> {
+            try {
```

Comment:
you don't have to use try-catch here - in case of other exceptions the test will fail. Assert.assertThrows uses Runnable with `throws Throwable`, so you don't have to wrap it

Reply:
1ef316b1557